crates/uv-distribution-types/src/build_requires.rs

```diff
@@ -23,16 +24,59 @@ impl std::ops::DerefMut for ExtraBuildRequires {
 }
 
 impl IntoIterator for ExtraBuildRequires {
-    type Item = (PackageName, Vec<Requirement>);
-    type IntoIter = std::collections::btree_map::IntoIter<PackageName, Vec<Requirement>>;
+    type Item = (PackageName, Vec<ExtraBuildRequirement>);
+    type IntoIter = std::collections::btree_map::IntoIter<PackageName, Vec<ExtraBuildRequirement>>;
 
     fn into_iter(self) -> Self::IntoIter {
         self.0.into_iter()
     }
 }
 
-impl FromIterator<(PackageName, Vec<Requirement>)> for ExtraBuildRequires {
-    fn from_iter<T: IntoIterator<Item = (PackageName, Vec<Requirement>)>>(iter: T) -> Self {
+impl FromIterator<(PackageName, Vec<ExtraBuildRequirement>)> for ExtraBuildRequires {
+    fn from_iter<T: IntoIterator<Item = (PackageName, Vec<ExtraBuildRequirement>)>>(
+        iter: T,
+    ) -> Self {
         Self(iter.into_iter().collect())
     }
 }
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct AnnotatedBuildRequirement {
+    /// The underlying [`Requirement`] for the build requirement.
+    pub requirement: Requirement,
+    /// Whether this build requirement should match the runtime environment.
+    pub match_runtime: bool,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum ExtraBuildRequirement {
```

Comment:
What about only having `AnnotatedBuildRequirement` with `match_runtime: Option<bool>`, to having to match on the two cases? This would allow us to `Either` too. The only place where we'd still need the enum is in the deserialize impl.

Reply:
That sounds reasonable.